strongbox-web-core/src/main/java/org/carlspring/strongbox/security/authentication/suppliers/NpmLoginAuthenticationSupplier.java

```diff
@@ -0,0 +1,80 @@
+package org.carlspring.strongbox.security.authentication.suppliers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.carlspring.strongbox.authentication.api.password.PasswordAuthentication;
+import org.carlspring.strongbox.controllers.layout.npm.NpmUser;
+import org.carlspring.strongbox.providers.layout.NpmLayoutProvider;
+
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import java.io.IOException;
+
+@Component
+public class NpmLoginAuthenticationSupplier
+        extends ArtifactAuthenticationSupplier
+{
+
+    @Inject
+    private ObjectMapper objectMapper;
+
+    public NpmLoginAuthenticationSupplier()
+    {
+        super(NpmLayoutProvider.ALIAS);
+    }
+
+    @CheckForNull
+    @Override
+    public Authentication supply(@Nonnull HttpServletRequest request)
+    {
+        NpmUser npmUser = deserializeNpmUser(request);
+
+        if (isValidNpmUser(npmUser))
```

Comment:
```
if (!isValidNpmUser(npmUser)) 
{
    throw new BadCredentialsException("invalid.credentials");
}
```

Reply:
Just to be clear, I need to invert my if-else?

Reply:
there will be no `else` in this case

Reply:
ah sorry. gotcha